config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -115,17 +115,15 @@ public void getConfig(
   public void getAllConfigs(
       GetAllConfigsRequest request, StreamObserver<GetAllConfigsResponse> responseObserver) {
     try {
-      List<ContextSpecificConfig> contextSpecificConfigList =
+      GetAllConfigsResponse getAllConfigsResponse =
           configStore.getAllConfigs(
               new ConfigResource(
                   request.getResourceName(), request.getResourceNamespace(), getTenantId()),
               request.getFilter(),
               request.getPagination(),
-              request.getSortByList());
-      responseObserver.onNext(
-          GetAllConfigsResponse.newBuilder()
-              .addAllContextSpecificConfigs(contextSpecificConfigList)
-              .build());
+              request.getSortByList(),
+              request.getIncludeTotal());
```

Comment:
So within the service, I would consider splitting up the total query - it doesn't really share anything with the config part. That way we're just adding a new API to the configStore, not adding branching and booleans that change behavior

Reply:
Fixed.